/home/temp1037/Duke/Tools/bwa.kit/bwa mem -aM -t 1 -R '@RG\tID:P39P.L002\tSM:P39P\tPL:ILLUMINA' /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta data/P39/P/L002-R1.fastq data/P39/P/L002-R2.fastq > data/P39/P/L002.sam
/home/temp1037/Duke/Tools/bwa.kit/bwa mem -aM -t 1 -R '@RG\tID:P39P.L001\tSM:P39P\tPL:ILLUMINA' /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta data/P39/P/L001-R1.fastq data/P39/P/L001-R2.fastq > data/P39/P/L001.sam
/home/temp1037/Duke/Tools/bwa.kit/bwa mem -aM -t 1 -R '@RG\tID:P39C.L003\tSM:P39C\tPL:ILLUMINA' /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta data/P39/C/L003-R1.fastq data/P39/C/L003-R2.fastq > data/P39/C/L003.sam
/home/temp1037/Duke/Tools/bwa.kit/bwa mem -aM -t 1 -R '@RG\tID:P39P.L003\tSM:P39P\tPL:ILLUMINA' /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta data/P39/P/L003-R1.fastq data/P39/P/L003-R2.fastq > data/P39/P/L003.sam
java -jar /home/temp1037/Duke/Tools/picard-tools-1.119/SortSam.jar I=data/P39/P/L002.sam O=data/P39/P/L002-sort.bam SO=coordinate
java -jar /home/temp1037/Duke/Tools/picard-tools-1.119/SortSam.jar I=data/P39/P/L001.sam O=data/P39/P/L001-sort.bam SO=coordinate
java -jar /home/temp1037/Duke/Tools/picard-tools-1.119/SortSam.jar I=data/P39/C/L003.sam O=data/P39/C/L003-sort.bam SO=coordinate
java -jar /home/temp1037/Duke/Tools/picard-tools-1.119/SortSam.jar I=data/P39/P/L003.sam O=data/P39/P/L003-sort.bam SO=coordinate
java -jar /home/temp1037/Duke/Tools/picard-tools-1.119/MarkDuplicates.jar I=data/P39/P/L002-sort.bam O=data/P39/P/L002-sort-dup.bam M=data/P39/P/L002-dup-metrics.txt
java -jar /home/temp1037/Duke/Tools/picard-tools-1.119/MarkDuplicates.jar I=data/P39/P/L001-sort.bam O=data/P39/P/L001-sort-dup.bam M=data/P39/P/L001-dup-metrics.txt
java -jar /home/temp1037/Duke/Tools/picard-tools-1.119/MarkDuplicates.jar I=data/P39/C/L003-sort.bam O=data/P39/C/L003-sort-dup.bam M=data/P39/C/L003-dup-metrics.txt
java -jar /home/temp1037/Duke/Tools/picard-tools-1.119/MarkDuplicates.jar I=data/P39/P/L003-sort.bam O=data/P39/P/L003-sort-dup.bam M=data/P39/P/L003-dup-metrics.txt
java -jar /home/temp1037/Duke/Tools/picard-tools-1.119/BuildBamIndex.jar INPUT=data/P39/P/L002-sort-dup.bam OUTPUT=data/P39/P/L002-sort-dup.bam.bai
java -jar /home/temp1037/Duke/Tools/picard-tools-1.119/BuildBamIndex.jar INPUT=data/P39/P/L001-sort-dup.bam OUTPUT=data/P39/P/L001-sort-dup.bam.bai
java -jar /home/temp1037/Duke/Tools/picard-tools-1.119/BuildBamIndex.jar INPUT=data/P39/C/L003-sort-dup.bam OUTPUT=data/P39/C/L003-sort-dup.bam.bai
java -jar /home/temp1037/Duke/Tools/picard-tools-1.119/BuildBamIndex.jar INPUT=data/P39/P/L003-sort-dup.bam OUTPUT=data/P39/P/L003-sort-dup.bam.bai
java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar -T RealignerTargetCreator -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/P/L002-sort-dup.bam --known /home/temp1037/Duke/Tools/data/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf -o data/P39/P/L002-realign.intervals
java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar -T RealignerTargetCreator -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/P/L001-sort-dup.bam --known /home/temp1037/Duke/Tools/data/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf -o data/P39/P/L001-realign.intervals
java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar -T RealignerTargetCreator -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/C/L003-sort-dup.bam --known /home/temp1037/Duke/Tools/data/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf -o data/P39/C/L003-realign.intervals
java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar -T RealignerTargetCreator -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/P/L003-sort-dup.bam --known /home/temp1037/Duke/Tools/data/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf -o data/P39/P/L003-realign.intervals
java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar -T IndelRealigner -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/P/L002-sort-dup.bam -known /home/temp1037/Duke/Tools/data/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf -targetIntervals data/P39/P/L002-realign.intervals -o data/P39/P/L002-sort-dup-realign.bam --filter_bases_not_stored
java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar -T IndelRealigner -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/P/L001-sort-dup.bam -known /home/temp1037/Duke/Tools/data/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf -targetIntervals data/P39/P/L001-realign.intervals -o data/P39/P/L001-sort-dup-realign.bam --filter_bases_not_stored
java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar -T IndelRealigner -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/C/L003-sort-dup.bam -known /home/temp1037/Duke/Tools/data/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf -targetIntervals data/P39/C/L003-realign.intervals -o data/P39/C/L003-sort-dup-realign.bam --filter_bases_not_stored
java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar -T IndelRealigner -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/P/L003-sort-dup.bam -known /home/temp1037/Duke/Tools/data/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf -targetIntervals data/P39/P/L003-realign.intervals -o data/P39/P/L003-sort-dup-realign.bam --filter_bases_not_stored
java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/P/L002-sort-dup-realign.bam -knownSites /home/temp1037/Duke/Tools/data/hg19/NA12878.HiSeq.WGS.bwa.cleaned.raw.subset.hg19.sites.vcf -knownSites /home/temp1037/Duke/Tools/data/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf -L /home/temp1037/Duke/Tools/data/S04380219_Covered.bed -o data/P39/P/L002-recal-table.txt
java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/P/L001-sort-dup-realign.bam -knownSites /home/temp1037/Duke/Tools/data/hg19/NA12878.HiSeq.WGS.bwa.cleaned.raw.subset.hg19.sites.vcf -knownSites /home/temp1037/Duke/Tools/data/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf -L /home/temp1037/Duke/Tools/data/S04380219_Covered.bed -o data/P39/P/L001-recal-table.txt
java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/C/L003-sort-dup-realign.bam -knownSites /home/temp1037/Duke/Tools/data/hg19/NA12878.HiSeq.WGS.bwa.cleaned.raw.subset.hg19.sites.vcf -knownSites /home/temp1037/Duke/Tools/data/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf -L /home/temp1037/Duke/Tools/data/S04380219_Covered.bed -o data/P39/C/L003-recal-table.txt
java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/P/L003-sort-dup-realign.bam -knownSites /home/temp1037/Duke/Tools/data/hg19/NA12878.HiSeq.WGS.bwa.cleaned.raw.subset.hg19.sites.vcf -knownSites /home/temp1037/Duke/Tools/data/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf -L /home/temp1037/Duke/Tools/data/S04380219_Covered.bed -o data/P39/P/L003-recal-table.txt
java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar -T PrintReads -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/P/L002-sort-dup-realign.bam -BQSR data/P39/P/L002-recal-table.txt -o data/P39/P/L002-recal.bam
java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar -T PrintReads -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/P/L001-sort-dup-realign.bam -BQSR data/P39/P/L001-recal-table.txt -o data/P39/P/L001-recal.bam
java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar -T PrintReads -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/C/L003-sort-dup-realign.bam -BQSR data/P39/C/L003-recal-table.txt -o data/P39/C/L003-recal.bam
/home/temp1037/Duke/Tools/bwa.kit/bwa mem -aM -t 1 -R '@RG\tID:P39C.L004\tSM:P39C\tPL:ILLUMINA' /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta data/P39/C/L004-R1.fastq data/P39/C/L004-R2.fastq > data/P39/C/L004.sam
java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar -T PrintReads -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/P/L003-sort-dup-realign.bam -BQSR data/P39/P/L003-recal-table.txt -o data/P39/P/L003-recal.bam
/home/temp1037/Duke/Tools/bwa.kit/bwa mem -aM -t 1 -R '@RG\tID:P39P.L004\tSM:P39P\tPL:ILLUMINA' /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta data/P39/P/L004-R1.fastq data/P39/P/L004-R2.fastq > data/P39/P/L004.sam
java -jar /home/temp1037/Duke/Tools/picard-tools-1.119/SortSam.jar I=data/P39/C/L004.sam O=data/P39/C/L004-sort.bam SO=coordinate
/home/temp1037/Duke/Tools/bwa.kit/bwa mem -aM -t 1 -R '@RG\tID:P39C.L002\tSM:P39C\tPL:ILLUMINA' /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta data/P39/C/L002-R1.fastq data/P39/C/L002-R2.fastq > data/P39/C/L002.sam
java -jar /home/temp1037/Duke/Tools/picard-tools-1.119/SortSam.jar I=data/P39/P/L004.sam O=data/P39/P/L004-sort.bam SO=coordinate
java -jar /home/temp1037/Duke/Tools/picard-tools-1.119/MarkDuplicates.jar I=data/P39/C/L004-sort.bam O=data/P39/C/L004-sort-dup.bam M=data/P39/C/L004-dup-metrics.txt
java -jar /home/temp1037/Duke/Tools/picard-tools-1.119/SortSam.jar I=data/P39/C/L002.sam O=data/P39/C/L002-sort.bam SO=coordinate
java -jar /home/temp1037/Duke/Tools/picard-tools-1.119/MarkDuplicates.jar I=data/P39/P/L004-sort.bam O=data/P39/P/L004-sort-dup.bam M=data/P39/P/L004-dup-metrics.txt
java -jar /home/temp1037/Duke/Tools/picard-tools-1.119/BuildBamIndex.jar INPUT=data/P39/C/L004-sort-dup.bam OUTPUT=data/P39/C/L004-sort-dup.bam.bai
java -jar /home/temp1037/Duke/Tools/picard-tools-1.119/MarkDuplicates.jar I=data/P39/C/L002-sort.bam O=data/P39/C/L002-sort-dup.bam M=data/P39/C/L002-dup-metrics.txt
java -jar /home/temp1037/Duke/Tools/picard-tools-1.119/BuildBamIndex.jar INPUT=data/P39/P/L004-sort-dup.bam OUTPUT=data/P39/P/L004-sort-dup.bam.bai
java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar -T RealignerTargetCreator -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/C/L004-sort-dup.bam --known /home/temp1037/Duke/Tools/data/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf -o data/P39/C/L004-realign.intervals
java -jar /home/temp1037/Duke/Tools/picard-tools-1.119/BuildBamIndex.jar INPUT=data/P39/C/L002-sort-dup.bam OUTPUT=data/P39/C/L002-sort-dup.bam.bai
java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar -T RealignerTargetCreator -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/P/L004-sort-dup.bam --known /home/temp1037/Duke/Tools/data/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf -o data/P39/P/L004-realign.intervals
java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar -T IndelRealigner -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/C/L004-sort-dup.bam -known /home/temp1037/Duke/Tools/data/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf -targetIntervals data/P39/C/L004-realign.intervals -o data/P39/C/L004-sort-dup-realign.bam --filter_bases_not_stored
java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar -T RealignerTargetCreator -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/C/L002-sort-dup.bam --known /home/temp1037/Duke/Tools/data/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf -o data/P39/C/L002-realign.intervals
java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar -T IndelRealigner -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/P/L004-sort-dup.bam -known /home/temp1037/Duke/Tools/data/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf -targetIntervals data/P39/P/L004-realign.intervals -o data/P39/P/L004-sort-dup-realign.bam --filter_bases_not_stored
java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/C/L004-sort-dup-realign.bam -knownSites /home/temp1037/Duke/Tools/data/hg19/NA12878.HiSeq.WGS.bwa.cleaned.raw.subset.hg19.sites.vcf -knownSites /home/temp1037/Duke/Tools/data/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf -L /home/temp1037/Duke/Tools/data/S04380219_Covered.bed -o data/P39/C/L004-recal-table.txt
java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar -T IndelRealigner -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/C/L002-sort-dup.bam -known /home/temp1037/Duke/Tools/data/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf -targetIntervals data/P39/C/L002-realign.intervals -o data/P39/C/L002-sort-dup-realign.bam --filter_bases_not_stored
java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/P/L004-sort-dup-realign.bam -knownSites /home/temp1037/Duke/Tools/data/hg19/NA12878.HiSeq.WGS.bwa.cleaned.raw.subset.hg19.sites.vcf -knownSites /home/temp1037/Duke/Tools/data/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf -L /home/temp1037/Duke/Tools/data/S04380219_Covered.bed -o data/P39/P/L004-recal-table.txt
java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar -T PrintReads -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/C/L004-sort-dup-realign.bam -BQSR data/P39/C/L004-recal-table.txt -o data/P39/C/L004-recal.bam
java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/C/L002-sort-dup-realign.bam -knownSites /home/temp1037/Duke/Tools/data/hg19/NA12878.HiSeq.WGS.bwa.cleaned.raw.subset.hg19.sites.vcf -knownSites /home/temp1037/Duke/Tools/data/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf -L /home/temp1037/Duke/Tools/data/S04380219_Covered.bed -o data/P39/C/L002-recal-table.txt
/home/temp1037/Duke/Tools/bwa.kit/bwa mem -aM -t 1 -R '@RG\tID:P39C.L001\tSM:P39C\tPL:ILLUMINA' /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta data/P39/C/L001-R1.fastq data/P39/C/L001-R2.fastq > data/P39/C/L001.sam
java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar -T PrintReads -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/P/L004-sort-dup-realign.bam -BQSR data/P39/P/L004-recal-table.txt -o data/P39/P/L004-recal.bam
java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar -T PrintReads -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/C/L002-sort-dup-realign.bam -BQSR data/P39/C/L002-recal-table.txt -o data/P39/C/L002-recal.bam
java -jar /home/temp1037/Duke/Tools/picard-tools-1.119/SortSam.jar I=data/P39/C/L001.sam O=data/P39/C/L001-sort.bam SO=coordinate
python /home/temp1037/bin/merge_lanes.py -e 'java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar' -T PrintReads -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/P/L001-recal.bam -I data/P39/P/L002-recal.bam -I data/P39/P/L003-recal.bam -I data/P39/P/L004-recal.bam --read_filter MappingQualityZero -o data/P39/P/merged.bam
java -jar /home/temp1037/Duke/Tools/picard-tools-1.119/MarkDuplicates.jar I=data/P39/C/L001-sort.bam O=data/P39/C/L001-sort-dup.bam M=data/P39/C/L001-dup-metrics.txt
java -jar /home/temp1037/Duke/Tools/picard-tools-1.119/MarkDuplicates.jar I=data/P39/P/merged.bam O=data/P39/P/merged-dup.bam M=data/P39/P/merged-dup-metrics.txt
java -jar /home/temp1037/Duke/Tools/picard-tools-1.119/BuildBamIndex.jar INPUT=data/P39/C/L001-sort-dup.bam OUTPUT=data/P39/C/L001-sort-dup.bam.bai
java -jar /home/temp1037/Duke/Tools/picard-tools-1.119/BuildBamIndex.jar INPUT=data/P39/P/merged-dup.bam OUTPUT=data/P39/P/merged-dup.bam.bai
java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar -T RealignerTargetCreator -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/C/L001-sort-dup.bam --known /home/temp1037/Duke/Tools/data/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf -o data/P39/C/L001-realign.intervals
java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar -T RealignerTargetCreator -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/P/merged-dup.bam --known /home/temp1037/Duke/Tools/data/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf -o data/P39/P/merged-realign.intervals
java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar -T IndelRealigner -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/C/L001-sort-dup.bam -known /home/temp1037/Duke/Tools/data/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf -targetIntervals data/P39/C/L001-realign.intervals -o data/P39/C/L001-sort-dup-realign.bam --filter_bases_not_stored
java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar -T IndelRealigner -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/P/merged-dup.bam -known /home/temp1037/Duke/Tools/data/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf -targetIntervals data/P39/P/merged-realign.intervals -o data/P39/P/merged-sort-dup-realign.bam --filter_bases_not_stored
java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/C/L001-sort-dup-realign.bam -knownSites /home/temp1037/Duke/Tools/data/hg19/NA12878.HiSeq.WGS.bwa.cleaned.raw.subset.hg19.sites.vcf -knownSites /home/temp1037/Duke/Tools/data/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf -L /home/temp1037/Duke/Tools/data/S04380219_Covered.bed -o data/P39/C/L001-recal-table.txt
java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/P/merged-sort-dup-realign.bam -knownSites /home/temp1037/Duke/Tools/data/hg19/NA12878.HiSeq.WGS.bwa.cleaned.raw.subset.hg19.sites.vcf -knownSites /home/temp1037/Duke/Tools/data/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf -L /home/temp1037/Duke/Tools/data/S04380219_Covered.bed -o data/P39/P/merged-recal-table.txt
java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar -T PrintReads -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/C/L001-sort-dup-realign.bam -BQSR data/P39/C/L001-recal-table.txt -o data/P39/C/L001-recal.bam
java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar -T PrintReads -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/P/merged-sort-dup-realign.bam -BQSR data/P39/P/merged-recal-table.txt -o data/P39/P/merged-recal.bam
python /home/temp1037/bin/merge_lanes.py -e 'java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar' -T PrintReads -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/C/L001-recal.bam -I data/P39/C/L002-recal.bam -I data/P39/C/L003-recal.bam -I data/P39/C/L004-recal.bam --read_filter MappingQualityZero -o data/P39/C/merged.bam
java -jar /home/temp1037/Duke/Tools/picard-tools-1.119/MarkDuplicates.jar I=data/P39/C/merged.bam O=data/P39/C/merged-dup.bam M=data/P39/C/merged-dup-metrics.txt
java -jar /home/temp1037/Duke/Tools/picard-tools-1.119/BuildBamIndex.jar INPUT=data/P39/C/merged-dup.bam OUTPUT=data/P39/C/merged-dup.bam.bai
java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar -T RealignerTargetCreator -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/C/merged-dup.bam --known /home/temp1037/Duke/Tools/data/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf -o data/P39/C/merged-realign.intervals
java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar -T IndelRealigner -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/C/merged-dup.bam -known /home/temp1037/Duke/Tools/data/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf -targetIntervals data/P39/C/merged-realign.intervals -o data/P39/C/merged-sort-dup-realign.bam --filter_bases_not_stored
java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar -T BaseRecalibrator -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/C/merged-sort-dup-realign.bam -knownSites /home/temp1037/Duke/Tools/data/hg19/NA12878.HiSeq.WGS.bwa.cleaned.raw.subset.hg19.sites.vcf -knownSites /home/temp1037/Duke/Tools/data/hg19/Mills_and_1000G_gold_standard.indels.hg19.sites.vcf -L /home/temp1037/Duke/Tools/data/S04380219_Covered.bed -o data/P39/C/merged-recal-table.txt
java -jar /home/temp1037/Duke/Tools/GenomeAnalysisTK.jar -T PrintReads -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -I data/P39/C/merged-sort-dup-realign.bam -BQSR data/P39/C/merged-recal-table.txt -o data/P39/C/merged-recal.bam
/usr/java/jdk1.6.0_16/bin/java -Xmx8g -jar /home/temp1037/Duke/Tools/MuTect/muTect-1.1.4.jar -T MuTect -R /home/temp1037/Duke/Tools/data/hg19/ucsc.hg19.fasta -cosmic /home/temp1037/Duke/Tools/data/hg19/Cosmic/Cosmic.hg19.vcf -dbsnp /home/temp1037/Duke/Tools/data/hg19/dbsnp_138.hg19.vcf --intervals chr17:1-7577200 --input_file:normal data/P39/C/merged-recal.bam --input_file:tumor data/P39/P/merged-recal.bam -o data/P39/muTect-out.txt -cov data/P39/muTect-coverage.txt
